src/routes/v2/collections.rs

```diff
@@ -0,0 +1,623 @@
+}
+#[get("collections")]
+pub async fn collections_get(
+    req: HttpRequest,
+    web::Query(ids): web::Query<CollectionIds>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<deadpool_redis::Pool>,
+    session_queue: web::Data<AuthQueue>,
+) -> Result<HttpResponse, ApiError> {
+    let ids = serde_json::from_str::<Vec<&str>>(&ids.ids)?;
+    let collections_data = database::models::Collection::get_many(&ids, &**pool, &redis).await?;
+
+    let user_option = get_user_from_headers(
+        &req,
+        &**pool,
+        &redis,
+        &session_queue,
+        Some(&[Scopes::COLLECTION_READ]),
+    )
+    .await
+    .map(|x| x.1)
+    .ok();
+
+    let collections = filter_authorized_collections(collections_data, &user_option, &pool).await?;
+
+    Ok(HttpResponse::Ok().json(collections))
+}
+
+#[get("{id}")]
+pub async fn collection_get(
+    req: HttpRequest,
+    info: web::Path<(String,)>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<deadpool_redis::Pool>,
+    session_queue: web::Data<AuthQueue>,
+) -> Result<HttpResponse, ApiError> {
+    let string = info.into_inner().0;
+
+    let collection_data = database::models::Collection::get(&string, &**pool, &redis).await?;
+    let user_option = get_user_from_headers(
+        &req,
+        &**pool,
+        &redis,
+        &session_queue,
+        Some(&[Scopes::COLLECTION_READ]),
+    )
+    .await
+    .map(|x| x.1)
+    .ok();
+
+    if let Some(data) = collection_data {
+        if is_authorized_collection(&data, &user_option, &pool).await? {
+            return Ok(HttpResponse::Ok().json(Collection::from(data)));
+        }
+    }
+    Ok(HttpResponse::NotFound().body(""))
+}
+
+//checks the validity of a project id or slug
+#[get("{id}/check")]
+pub async fn collection_get_check(
+    info: web::Path<(String,)>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<deadpool_redis::Pool>,
+) -> Result<HttpResponse, ApiError> {
+    let slug = info.into_inner().0;
+
+    let collection_data = database::models::Collection::get(&slug, &**pool, &redis).await?;
+
+    if let Some(collection) = collection_data {
+        Ok(HttpResponse::Ok().json(json! ({
+            "id": models::ids::CollectionId::from(collection.id)
+        })))
+    } else {
+        Ok(HttpResponse::NotFound().body(""))
+    }
+}
+
+#[derive(Deserialize, Validate)]
+pub struct EditCollection {
+    #[validate(
+        length(min = 3, max = 64),
+        custom(function = "crate::util::validate::validate_name")
+    )]
+    pub title: Option<String>,
+    #[validate(length(min = 3, max = 256))]
+    pub description: Option<String>,
+    #[validate(length(max = 65536))]
+    pub body: Option<String>,
+    #[validate(
+        length(min = 3, max = 64),
+        regex = "crate::util::validate::RE_URL_SAFE"
+    )]
+    pub slug: Option<String>,
+    #[validate(length(max = 64))]
+    pub add_projects: Option<Vec<String>>,
+    #[validate(length(max = 64))]
+    pub remove_projects: Option<Vec<String>>,
+}
```

Comment:
this could just take a list of project ids (see the categories edit for projects) instead of 2 arrays